terraform-modules/gcs_bq_log_sink/log_sinks.tf

```diff
@@ -61,6 +61,35 @@ resource "google_logging_project_sink" "bucket-log-sink" {
   filter                 = var.log_filter
   unique_writer_identity = true
 }
+######
+
+resource "google_pubsub_topic" "pubsub" {
+  count   = var.enable_pubsub
+  name    = "${var.application_name}-${var.owner}-logs-pubsub${var.nonce != "" ? "_${var.nonce}" : ""}"
+}
+
+# Because our sink uses a unique_writer, we must grant that writer access to the bucket.
+resource "google_project_iam_binding" "log-writer" {
+    role = "roles/pubsub.admin"
+    count       = var.enable_pubsub
+    members = [
+        "${google_logging_project_sink.pubsub-log-sink[0].writer_identity}",
+    ]
+}
+
+# Our sink; this logs all activity; This requires your SA to have Logging/Logs Configuration Writer 
+resource "google_logging_project_sink" "pubsub-log-sink" {
+    count       = var.enable_pubsub
+    name        = "${var.application_name}-${var.owner}-pubsub-log-sink${var.nonce != "" ? "_${var.nonce}" : ""}"
+    destination = "pubsub.googleapis.com/projects/${var.project}/topics/${google_pubsub_topic.pubsub[0].name}"
+    filter      = var.log_filter
+    unique_writer_identity = true
+}
+
+# Outputs
+output "pubsub_writer_identity" {
+  value = google_logging_project_sink.pubsub-log-sink.*.writer_identity
```

Comment:
This `*` makes it work even when you don't enable it, right?

Reply:
The other outputs show the unique writer so i did the same. If not enabled it should be an empty string.